src/index.jsx

```diff
@@ -1,17 +1,21 @@
-import React from "react";
-import ReactDOM from "react-dom";
-import "./index.css";
-import App from "./App";
-import reportWebVitals from "./reportWebVitals";
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+
+import rootReducer from './app/rootReducer';
+import App from './app';
+import './index.css';
+
+const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
+const rootElement = document.getElementById('root');
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <Provider store={createStoreWithMiddleware(rootReducer)}>
```

Comment:
When using redux, another pretty useful and simple middleware is [redux-logger](https://www.npmjs.com/package/redux-logger). 
It allows you to check the previous state and the resultant state of every action dispatched.
Check it out if you don't know it yet!